### PR TITLE
add ecal rechit error code to rechitanalyzer

### PR DESCRIPTION
**Pull request by bi-ran**
Describe the Pull-Request: Add ecal rechit error code to rechitanalyzer.

Does the PR pass the test script located at
HeavyIonsAnalysis/JetAnalysis/test/runtest.sh
[X] Yes
[] No

Please make sure to mention (using the @ syntax) anyone who might be interested in this PR.
@richard-cms

### CODE REVIEW

**Comment by harmonicoscillator** [comment]
@bi-ran I think that the datatype for flags should be uint32, not int. Maybe it's okay since we're running on 64 bit systems already? I'll comment inline.

File: HeavyIonsAnalysis/JetAnalysis/src/RecHitTreeProducer.cc [hunk 1]
```diff
@@ -98,6 +98,8 @@ struct MyRecHit{
   float hadEt[MAXHITS];
   float chi2[MAXHITS];
   float eError[MAXHITS];
+
+  int flags[MAXHITS];
```
Comment (harmonicoscillator) [inline HeavyIonsAnalysis/JetAnalysis/src/RecHitTreeProducer.cc]: I think this should be uint32

File: HeavyIonsAnalysis/JetAnalysis/src/RecHitTreeProducer.cc [hunk 2]
```diff
@@ -876,6 +890,7 @@ RecHitTreeProducer::beginJob()
     eeTree->Branch("perp",eeRecHit.perp,"perp[n]/F");
     eeTree->Branch("chi2",eeRecHit.chi2,"chi2[n]/F");
     eeTree->Branch("eError",eeRecHit.eError,"eError[n]/F");
+    eeTree->Branch("flags",eeRecHit.flags,"flags[n]/I");
```
Comment (harmonicoscillator) [inline HeavyIonsAnalysis/JetAnalysis/src/RecHitTreeProducer.cc]: I think that "flags[n]/I" should instead be "flags[n]/i" (lowercase 'i' for unsigned).

File: HeavyIonsAnalysis/JetAnalysis/src/RecHitTreeProducer.cc [hunk 3]
```diff
@@ -888,6 +903,7 @@ RecHitTreeProducer::beginJob()
     ebTree->Branch("perp",ebRecHit.perp,"perp[n]/F");
     ebTree->Branch("chi2",ebRecHit.chi2,"chi2[n]/F");
     ebTree->Branch("eError",ebRecHit.eError,"eError[n]/F");
+    ebTree->Branch("flags",ebRecHit.flags,"flags[n]/I");
```
Comment (harmonicoscillator) [inline HeavyIonsAnalysis/JetAnalysis/src/RecHitTreeProducer.cc]: lowercase i again

**Comment by bi-ran** [comment]
@richard-cms i've updated the var types to match

**Comment by harmonicoscillator** [comment]
Great, let me run a short test and then I'll merge.

**Pull request merged**